framework/scripts/tests/test_cluster_control.py

```diff
@@ -0,0 +1,351 @@
+                                      f"{worker_status['last_keep_alive']}.\n            Integrity check:\n           "
+                                      f"     Last integrity check: n/a "
+                                      f"({worker_status['last_check_integrity']['date_end_master']} - "
+                                      f"{worker_status['last_check_integrity']['date_start_master']}).\n               "
+                                      f" Permission to check integrity: {worker_status['sync_integrity_free']}.\n      "
+                                      f"      Integrity sync:\n                Last integrity synchronization: 1.0s "
+                                      f"({worker_status['last_sync_integrity']['date_start_master']} - "
+                                      f"{worker_status['last_sync_integrity']['date_end_master']}).\n                "
+                                      f"Synchronized files: Shared: "
+                                      f"{worker_status['last_sync_integrity']['total_files']['shared']} | Missing: "
+                                      f"{worker_status['last_sync_integrity']['total_files']['missing']} | Extra: "
+                                      f"{worker_status['last_sync_integrity']['total_files']['extra']} | Extra valid: "
+                                      f"{worker_status['last_sync_integrity']['total_files']['extra_valid']}.\n     "
+                                      f"           Extra valid files correctly updated in master: "
+                                      f"{worker_status['last_sync_integrity']['total_extra_valid']}.\n            "
+                                      f"Agents-info:\n                Last synchronization: 0.001s ("
+                                      f"{worker_status['last_sync_agentinfo']['date_start_master']} - "
+                                      f"{worker_status['last_sync_agentinfo']['date_start_master']}).\n         "
+                                      f"       Number of synchronized chunks: "
+                                      f"{worker_status['last_sync_agentinfo']['n_synced_chunks']}.\n                "
+                                      f"Permission to synchronize agent-info: {worker_status['sync_agent_info_free']}"
+                                      f".\n")])
+
+    local_client_mock.assert_called_once_with()
+    get_health_mock.assert_called_once_with(local_client_mock.return_value, filter_node=filter_node)
+    get_utc_strptime_mock.assert_has_calls(
+        [call(worker_status['last_sync_integrity']['date_end_master'], '%Y-%m-%dT%H:%M:%S.%fZ'),
+         call(worker_status['last_sync_integrity']['date_start_master'], '%Y-%m-%dT%H:%M:%S.%fZ'),
+         call(worker_status['last_sync_agentinfo']['date_end_master'], '%Y-%m-%dT%H:%M:%S.%fZ'),
+         call(worker_status['last_sync_agentinfo']['date_start_master'], '%Y-%m-%dT%H:%M:%S.%fZ')])
```

Comment:
We need to wait for #10771 to close to add the relevant checks for agent-groups.